packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js

```diff
@@ -1,83 +1,170 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import memoize from 'lodash/memoize';
+import { chunk } from 'lodash';
+import format from 'date-fns/format';
+import endOfMonth from 'date-fns/end_of_month';
+import startOfWeek from 'date-fns/start_of_week';
+import getDate from 'date-fns/get_date';
+import setDay from 'date-fns/set_day';
+import getMonth from 'date-fns/get_month';
+import isSameDay from 'date-fns/is_same_day';
+import differenceInCalendarWeeks from 'date-fns/difference_in_calendar_weeks';
+
+import addDays from 'date-fns/add_days';
 import theme from './DatePicker.scss';
 import DayPickerAction from './DayPickerAction';
 
-function DatePicker(props) {
-	const dayNames = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
-	const days = (new Array(7))
-					.fill(0)
-					.map((_, i) => i + 1);
+const FIRST_DAY_OF_WEEK = 1;
+const BASE_DATE = new Date(0);
```

Comment:
If I let open my app at 23h50 and I have a re-render at 00h01 ?

Reply:
Yes i forgot to ask the question, it was in my head too.
But it's not this line the problem it's in the DateTimeView.
(new Date(0) => 0ms from Date 1970-01-01T00:00:00.000Z, so not relative to current time)

```javascript
class DateTimeView extends React.Component {

	constructor(props) {
		super(props);

		this.state = {
			today: new Date(),
		};
```

Do I need to recreate the today date at each render ?
Or i set a timeout to update the today date in state when we are changing day to keep the render pure ?

Reply:
Check if today directly in the picker which is no more pure.
Need a re-render to update the visual today day, not re-render if staying in front of the calendar while there is a date changed. (23:59 - 00:00)

Reply:
✔